fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1365,7 +1348,9 @@ def forward(self, *args: Any, **kwargs: Any) -> torch.Tensor:
     @torch.no_grad()
     def _free_ssd_offload(self) -> None:
         if self.ssd_offload:
-            self.ssd_buffer.to_disk()
+            for p in self.params:
+                assert isinstance(p, SsdFlatParameter)
+                p.to_file(permit_when_tensor_none=True)
```

Comment:
why do we need the permit_when_tensor_none bool? what condition requires this check?

Reply:
In order to make _free_ssd_offload idempotent, we need to make sure that if this function is called when the SsdFlatParam is already offloaded to disk, no error occurs. 

Without `permit_when_tensor_none` set to True, calling p.to_file() when the SsdFlatParam is already offloaded to disk results in an error.